torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -1020,6 +1021,28 @@ const Expr* PolynomialTransformer::mutate(const Mod* v) {
     return getImmediateByType(v->dtype(), 0);
   }
 
+  // x % C => x, if x is known in [0, C), e.g. x is a loop index
+  // with constant
+  if (rhs_new->isConstant()) {
```

Comment:
Ideally, this should work not only for constants. E.g. if a loop goes to `N`, then it would be good if we can simplify `i % N` as well. If we don't do this, then we're in a danger of falling from a performance cliff when we switch from static to dynamic shapes.

Supporting non-constant bounds might require much more work and probably would need a proper bound inference for variables and I realize that it might be worthwhile starting with a simple case first. If you prefer that, please leave a TODO and/or file an issue on that so that we do not forget about it.